File: Examples/DotNET/SDK/CSharp/Images/UpdateGifProperties.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Imaging. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////

using System;
using Aspose.Cloud;

namespace Aspose.Imaging.Cloud.Examples.Images
{
    class UpdateGifProperties
    {
        static void Main()
        {
            string dataDir = Common.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
            string input = "sample1.gif";
            string output = "output.gif";

            Common.StorageService.File.UploadFile(dataDir + input, input, storage: Common.STORAGE );

            byte backgroundColorIndex = 255;
            byte colorResolution = 7;
            bool hasTrailer = true;
            bool interlaced = true;
            bool isPaletteSorted = true;
            byte pixelAspectRatio = 10;
            string outPath = "Imaging/"+input;

            Common.GetImagingSdk().Gif.UpdateParametersOfGifImage(
                input,
                backgroundColorIndex,
                colorResolution,
                hasTrailer,
                interlaced,
                isPaletteSorted,
                pixelAspectRatio,
                outPath,
                Common.FOLDER,
                storage: Common.STORAGE
                );

            Common.StorageService.File.DownloadFile(outPath, dataDir+output, storage: Common.STORAGE);

        }
    }
}

File: Examples/PHP/SDK/images/UpdateGifProperties.php
<?php require "../common.php";

$inputName = "sample1.gif";
$inputPath = getPath(__FILE__, $inputName);
$outputName = "output.gif";

$backgroundColorIndex = 255;
$colorResolution = 100;
$hasTrailer = true;
$interlaced = true;
$isPaletteSorted = true;
$pixelAspectRatio = 10;
$fromScratch = false;

$outputPath = getPath(__FILE__, $outputName);

getStorageSdk()->PutCreate($inputName, NULL, NULL, $inputPath);

$response = getImagingSdk()->GetImageGif($inputName, $backgroundColorIndex, $colorResolution, $hasTrailer, $interlaced, $isPaletteSorted, $pixelAspectRatio, $fromScratch, $outputPath, getFolder(), getStorage());;
$result = $storageApi->GetDownload($inputName);
getApiClient()->saveFile($result, $outputPath, $inputName);


